### ISSUE TITLE
Button presses have slight delay

### ISSUE THREAD

**Issue by spuder** [spuder/MuteMe-arduino#2]
The button _feels_ slow to respond. Most likely this is because we are using polling to detect if the user is holding the button. 

Theories
1. Serial.println is taking too long

I've tried both debug and non debug builds (non debug builds don't have any serial.println statements since they are wrapped in if statements). Both _feel_ slow

2. Debounce is too long or too short

I've tried values from 30 up to 250

```
  if (millis() - buttonCheckMillis > 50) {
    buttonCheckMillis = millis();
```

Shorter values _feel_ less responsive. Maybe because of all the serial.println? 
Longer values like 250 didn't seem to help either.

I looked at how other button libraries handle debouncing and they do values between 30ms and 60ms

```
void loop()
{
  // User pressed the button
  if (button1.toggled()) {
    if (button1.read() == Button::PRESSED) {
        #ifdef DEBUG
          Serial.println("Button Pressed");
        #endif
        rawhidData[3] = 0x04; // Touch Button
        RawHID.write(rawhidData, sizeof(rawhidData));
        rawhidData[3] = 0x00;
    }
    // User released the button
    else {
        #ifdef DEBUG
          Serial.println("Button Released");
        #endif
        rawhidData[3] = 0x02; // Release Button
        RawHID.write(rawhidData, sizeof(rawhidData));
        rawhidData[3] = 0x00;
    }
  }
  // Poll every 50 milliseconds to see if user is still holding button
  // TODO: This is a little slow. Find a way to make it more responsive.
  if (millis() - buttonCheckMillis > 50) {
    buttonCheckMillis = millis();
    if (button1.read() == Button::PRESSED) {
        #ifdef DEBUG
          Serial.println("Button Held");
        #endif
        rawhidData[3] = 0x01; // Hold Button
        RawHID.write(rawhidData, sizeof(rawhidData));
        rawhidData[3] = 0x00;
    }
  }
```

**Comment by spuder** [comment]
Here is an output of pushing the button quickly. The button is still showing as pressed for several hundred milliseconds

```
Ignoring data61
21 
61 
Ignoring data61
21 
61 
Ignoring data61
21 
61 
Ignoring data61
Button Pressed
Button Held
Button Held
Button Held
Button Held
Button Held
Button Held
Button Held
Button Held
Button Held
Button Held
Button Released
4 
44 
Ignoring data44
21 
61 
```

**Comment by spuder** [comment]
Fixed in 6af001c2986b8d52ebef22e524056129510b8912

switched to Button2 library with callback functions. Much snappy now
